packages/filesystem/src/browser/file-tree/file-tree-widget.ts

```diff
@@ -56,4 +63,108 @@ export class FileTreeWidget extends TreeWidget {
+                ondrop: event => this.handleDropEvent(node, event, true),
+            };
+        }
+        return elementAttrs;
+    }
+
+    protected handleDragEnterEvent(node: ITreeNode, event: DragEvent): void {
+        event.preventDefault();
+        event.stopPropagation();
+        const target = event.target as HTMLElement;
+        if (target.parentElement) {
+            target.parentElement.classList.add(activeDropZone);
+        }
+    }
+
+    protected handleDragOverEvent(node: ITreeNode, event: DragEvent): void {
+        event.preventDefault();
+    }
+
+    protected handleLeaveEvent(event: DragEvent): void {
+        event.preventDefault();
+        event.stopPropagation();
+        const target = event.target as HTMLElement;
+        if (target.parentElement) {
+            target.parentElement.classList.remove(activeDropZone);
+        }
+    }
+
+    protected handleDropEvent(node: ITreeNode, event: DragEvent, dropAllowed: boolean): void {
+        event.stopPropagation();
+        event.preventDefault();
+        if (dropAllowed) {
+            event.dataTransfer.dropEffect = 'copy'; // Explicitly show this is a copy.
+            const target = event.target as HTMLElement;
+            if (target.parentElement) {
+                target.parentElement.classList.remove(activeDropZone);
+            }
+            this.dropHandler(event, node);
+        }
+    }
+
+    protected dropHandler(ev: DragEvent, node: ITreeNode) {
+        ev.preventDefault();
+        const data = ev.dataTransfer.items;
+
+        for (let i = 0; i < data.length; i += 1) {
+            if (data[i].kind === 'file') {
+                const entry: WebKitEntry = data[i].webkitGetAsEntry();
+
+                if (entry.isDirectory) {
+                    this.handleDirectory(entry, node.id);
+                } else {
+                    const f: any = data[i].getAsFile();
+                    this.model.upload(node.id, f, f.name);
+                }
+            }
+        }
+    }
+
+    protected handleDirectory(webDirEntry: WebKitEntry, dest: string) {
+
+        const getEntries = (dirEntry: any, destination: string) => {
+            const reader = dirEntry.createReader();
+            reader.readEntries((e: any) => {
+                const results = <WebKitEntry[]>e;
+
+                for (let count = 0; count < results.length; count++) {
+
+                    if (results[count].isDirectory) {
+                        getEntries(results[count], destination);
+                    } else {
+                        const fileEntry = <WebKitFileEntry>results[count];
+                        fileEntry.file(e => {
+                            const f = <File>(e as {});
+                            this.model.upload(destination + dirEntry.fullPath, f, f.name);
+                        });
+                    }
+                }
+            }, (error: any) => {
+                this.logger.error(error.toString());
```

Comment:
just pass error: `this.logger.error(error)`

Reply:
Done